### PR TITLE
optimise system image using simg2dontcare

### PR DESCRIPTION
**Pull request by incognitojam**
Generate an additional "optimized" image for any sparse partition. Adds `url_optimized` and `hash_optimized` properties to manifest.

As the optimized image requires the partition to be erased first, the original image is preserved and still uploaded so that we don't break existing flash/openpilot updater.

split out from https://github.com/commaai/agnos-builder/pull/165

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Once this is merged, can you also clean up provisioning to use this?

File: internal/package_ota.sh [hunk 1]
```diff
@@ -70,6 +70,31 @@ EOF
     "url": "$AGNOS_STAGING_UPDATE_URL/$FILENAME"
 EOF
 
+  if [ "$SPARSE" == "true" ]; then
```
Comment (adeebshihadeh) [inline internal/package_ota.sh:73]: I thought we talked about making this a separate script that takes an agnos manifest?

Comment (incognitojam) [reply internal/package_ota.sh:73]: I can still write that script (to be able to process the existing manifest?), but I wanted to check whether we need to have the output in a separate manifest or not - I didn't know what that helped with

Comment (adeebshihadeh) [reply internal/package_ota.sh:73]: it has to be a different manifest since openpilot's agnos updater flashes all the entries in the manifest

Comment (incognitojam) [reply internal/package_ota.sh:73]: do we want openpilot to use this image eventually?

Comment (adeebshihadeh) [reply internal/package_ota.sh:73]: no

**Comment by incognitojam** [comment]
I don't think this needs to be split out anymore https://github.com/commaai/agnos-builder/pull/165/files

**Pull request closed without merging**